pandas/io/html.py

```diff
@@ -924,22 +928,22 @@ def _parse(flavor, io, match, attrs, encoding, displayed_only, **kwargs):
 
 @deprecate_nonkeyword_arguments(version="2.0")
 def read_html(
-    io,
-    match=".+",
-    flavor=None,
+    io: FilePathOrBuffer,
+    match: Union[str, Pattern] = ".+",
+    flavor: Optional[str] = None,
     header=None,
     index_col=None,
     skiprows=None,
```

Comment:
The above params are list-likes, which are not expressable in the type system, AFAIK.

Reply:
I think fine to use something like `Optional[Union[int, Sequence[int]]`